### PR TITLE
Fixed #343 Control EXECUTOR_THREAD_POOL_SIZE

### LINKED ISSUE #343: Control EXECUTOR_THREAD_POOL_SIZE
https://github.com/couchbase/couchbase-lite-java-core/blob/master/src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java#L72

The number of threads should be a parameter when a replicaiton object is created.

One reason is that I noticed that each thread makes a PUT request (_local/...) when it's done, to save the lastSequence.
This may be a costly overhead, in case you don't care about the replication speed and the documents are fetched in bulks (using _all_docs or _bulk_get).

### PR DESCRIPTION
**Pull request by hideki**
- Make executor_thread_pool_size for RemoteRequestExecutor to be configurable

### CODE REVIEW

**Comment by pasin** [comment]
As this sort of changes the public API, can you review this with @zgramana?

File: src/main/java/com/couchbase/lite/ManagerOptions.java [hunk 1]
```diff
@@ -6,15 +6,22 @@
...
+     * automatically migrate blobstore filename
      */
     private boolean autoMigrateBlobStoreFilename = false;
 
+    /**
+     * the number of threads to keep in the pool for RemoteRequestExecutor
+     * ReplicationInternal.java: public static final int EXECUTOR_THREAD_POOL_SIZE = 5;
+     * https://github.com/couchbase/couchbase-lite-java-core/issues/343
+     */
+    private int executorThreadPooSize = -1;
```
Comment (zgramana) [inline src/main/java/com/couchbase/lite/ManagerOptions.java]: Spelling error in 'Poo'. And is `-1` the right default here?

Comment (hideki) [inline src/main/java/com/couchbase/lite/ManagerOptions.java]: - thanks for catching my stupid spelling error.
- If 0 or lower, CBL use default value. 0 could be better as default number.

**Comment by hideki** [comment]
@zgramana Could you please review again?

**Comment by zgramana** [comment]
:+1:

**Comment by hideki** [comment]
@zgramana This PR is unable to automatic merge. Can I merge after resolving conflict by myself?

**Pull request merged**